x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/tags/read_tags_route.ts

```diff
@@ -4,49 +4,40 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import Hapi from 'hapi';
-
+import { IRouter } from '../../../../../../../../../src/core/server';
 import { DETECTION_ENGINE_TAGS_URL } from '../../../../../common/constants';
-import { LegacyServices, LegacyRequest } from '../../../../types';
-import { transformError } from '../utils';
+import { transformError, buildSiemResponse } from '../utils';
 import { readTags } from '../../tags/read_tags';
-import { GetScopedClients } from '../../../../services';
 
-export const createReadTagsRoute = (getClients: GetScopedClients): Hapi.ServerRoute => ({
-  method: 'GET',
-  path: DETECTION_ENGINE_TAGS_URL,
-  options: {
-    tags: ['access:siem'],
-    validate: {
+export const readTagsRoute = (router: IRouter) => {
+  router.get(
+    {
+      path: DETECTION_ENGINE_TAGS_URL,
+      validate: false,
       options: {
-        abortEarly: false,
+        tags: ['access:siem'],
       },
     },
-  },
-  async handler(request: LegacyRequest, headers) {
-    const { alertsClient } = await getClients(request);
+    async (context, request, response) => {
+      const alertsClient = context.alerting.getAlertsClient();
+      const siemResponse = buildSiemResponse(response);
 
-    if (!alertsClient) {
-      return headers.response().code(404);
-    }
+      if (!alertsClient) {
+        return siemResponse.error({ statusCode: 404 });
+      }
```

Comment:
For parts of the code like this I see what looks like alertsClient can no longer be null/undefined?

<img width="465" alt="Screen Shot 2020-03-02 at 11 08 16 PM" src="https://user-images.githubusercontent.com/1151048/75747658-c7076d00-5cda-11ea-9ef6-3501e914ebda.png">

If so, it would be nice to remove boiler plate checks for it be non-existent? If it still can be, I question the type there and why it doesn't reflect it can be null. But overall I'm hopeful we can remove more of the boiler plate like the checks for alertsClient to exst.

Reply:
If SIEM can be run without alerting, then the check is still valid.

A final step of the NP migration is going to be filling out our `kibana.json` with our optional and required dependencies and updating our types appropriately. I've been updating [our typings](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/siem/server/plugin.ts#L40-L51) as I discover dependencies, so I think we've got a comprehensive list there. My hope is that NP does some magic to make these context types dynamic based on your dependencies, but I have not seen that behavior thus far.

Reply:
@FrankHassanabad correction: the alerting context type should be optional here, so that needs to be fixed upstream.